exporter/loadbalancingexporter/resolver_k8s_handler.go

```diff
@@ -0,0 +1,102 @@
+package loadbalancingexporter
+
+import (
+	"context"
+	"sync"
+
+	"go.uber.org/zap"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/client-go/tools/cache"
+)
+
+var _ cache.ResourceEventHandler = (*handler)(nil)
+
+type handler struct {
+	endpoints *sync.Map
+	callback  func(ctx context.Context) ([]string, error)
+	logger    *zap.Logger
+}
+
+func (h handler) OnAdd(obj interface{}, isInInitialList bool) {
+	h.logger.Debug("onAdd called")
+	var endpoints []string
+
+	switch object := obj.(type) {
+	case *corev1.Endpoints:
+		endpoints = convertToEndpoints(object)
+	default: // unsupported
+		h.logger.Warn("onAdd unable to handle object", zap.Any("obj", obj))
```

Comment:
It would be best to record the object's type, rather than the object itself.

Reply:
Thanks, that's much better, will adjust accordingly.